### DISCUSSION TITLE
Custom Client Session Handling redirect wont work -  unauthorized: ""

### DISCUSSION THREAD

**Discussion by RadasLove** [nextauthjs/next-auth#4376] [Help] [answered]
### Question 💬

Hey, i have one issue, i need redirect page but wont work

I want use Custom Client Session Handling
From https://next-auth.js.org/getting-started/client

Any redirect in any of this object wont work

```
IndexPage.auth = {
  unauthorized: '/auth/my-login-page'' // redirect to this url
};
```
```
_app.tsx

export default function App({
  Component,
  pageProps: { session, ...pageProps },
}) {
  return (
    <SessionProvider session={session}>
      {Component.auth ? (
        <Auth>
          <Component {...pageProps} />
        </Auth>
      ) : (
        <Component {...pageProps} />
      )}
    </SessionProvider>
  );
}

function Auth({ children }) {

  const { status } = useSession({ required: true });

  if (status === 'loading') {
    return <div>Loading...</div>;
  }

  return children;
}
```

```
[..nextuath].js

  pages: {
    signIn: '/auth/login',
  },

```
Protecting page work, but the default redirect work on signIn page but i want somewhere different links. or have default signIn withou query error parameter
"?callbackUrl=http%3A%2F%2Flocalhost%3A3001%2Fakce%2Fvytvorit&error=SessionRequired"

Have somebody some experience?

### How to reproduce ☕️

 https://next-auth.js.org/getting-started/client

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by ThangHuuVu** [comment, marked as answer]
Custom Client Session Handling is not a NextAuth.js API, so you'd need to implement your own redirection.
```js
function Auth({ children, unauthorized }) {
  const { status } = useSession()
  if (status === 'unauthenticated') {
    // navigate to `unauthorized`
  }
  
  return children
}
```

**Reply by lsbyerley** [reply]
@ThangHuuVu is this question related to your answer? https://github.com/nextauthjs/next-auth/discussions/4376

**Reply by lsbyerley** [reply]
@ThangHuuVu sorry i posted the wrong link.. i mean this question https://github.com/nextauthjs/next-auth/discussions/4366

**Comment by vnugent** [comment]
A follow up question.  How do you direct users to the login page the reactive way?  Such as in your example, when a user is trying to access a protected page, direct them to the login page.

I understand there are signIn() and signOut() API that you can call imperatively when users click on a button or a link.  


```
function Auth({ children, unauthorized }) {
  const { status } = useSession()
  if (status === 'unauthenticated') {
    // navigate to `unauthorized` 
   // I want to direct user to the log in page here.  But how?
   
  }
  
  return children
}
```

**Reply by ThangHuuVu** [reply]
@vnugent You can use the [Next.js router](https://nextjs.org/docs/api-reference/next/router#userouter) on the client side:
```js
function Auth({ children, unauthorized }) {
  const { status } = useSession()
  const router = useRouter();
  if (status === 'unauthenticated') {
    // navigate to `unauthorized` 
    router.push("/login");
    return;
  }
  return children
}
```

**Comment by getsnoopy** [comment]
Another follow up question to this pattern: doing `Component.admin` in `_app.tsx` results in this type error:
```
Property 'auth' does not exist on type 'ComponentType<{}>'.
  Property 'auth' does not exist on type 'ComponentClass<{}, any>'.
```

How does one create a custom `ComponentClass` or `ComponentType`?
